aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -397,11 +397,20 @@ public void confirmSignIn(
             @NonNull Consumer<AuthException> onException
     ) {
         final Map<String, String> metadata = new HashMap<>();
+        final Map<String, String> userAttributes = new HashMap<>();
         if (options instanceof AWSCognitoAuthConfirmSignInOptions) {
-            metadata.putAll(((AWSCognitoAuthConfirmSignInOptions) options).getMetadata());
+            if (((AWSCognitoAuthConfirmSignInOptions) options).getMetadata() != null) {
+                metadata.putAll(((AWSCognitoAuthConfirmSignInOptions) options).getMetadata());
+            }
+
+            if (((AWSCognitoAuthConfirmSignInOptions) options).getUserAttributes() != null) {
+                for (AuthUserAttribute attribute : ((AWSCognitoAuthConfirmSignInOptions) options).getUserAttributes()) {
+                    userAttributes.put(attribute.getKey().getKeyString(), attribute.getValue());
+                }
+            }
```

Comment:
null check isn't necessary because `AWSCognitoAuthConfirmSignInOptions` returns non-null for both `getMetadata()` and `getUserAttributes()`.

Reply:
Done.